src/components/CollectionProvider/CollectionProvider.tsx

```diff
@@ -1,28 +1,61 @@
 import * as React from 'react'
-import { Props } from './CollectionProvider.types'
+import equal from 'fast-deep-equal'
+import { DEFAULT_ITEMS_PAGE_SIZE, DEFAULT_ITEMS_PAGE, Props } from './CollectionProvider.types'
 
 export default class CollectionProvider extends React.PureComponent<Props> {
+  fetchCollectionItems(itemsPage: number | number[] = DEFAULT_ITEMS_PAGE) {
+    const { id, onFetchCollectionItems, itemsPageSize } = this.props
+    if (id) {
+      onFetchCollectionItems(id, itemsPage, itemsPageSize || DEFAULT_ITEMS_PAGE_SIZE)
+    }
+  }
+
   componentDidMount() {
-    const { id, onFetchCollection, isConnected } = this.props
+    const { id, onFetchCollection, isConnected, itemsPage } = this.props
     if (id && isConnected) {
       onFetchCollection(id)
+      this.fetchCollectionItems(itemsPage)
     }
   }
 
   componentDidUpdate(prevProps: Props) {
-    const { id, onFetchCollection, isConnected, collection } = this.props
+    const { id, isConnected, collection, itemsPage, onFetchCollection } = this.props
     const justFinishedConnecting = !prevProps.isConnected && isConnected
     if (id && justFinishedConnecting) {
       onFetchCollection(id)
+      this.fetchCollectionItems(itemsPage)
     }
 
     if (id && id !== prevProps.id && !collection) {
       onFetchCollection(id)
     }
+
+    // logic to fetch the new pages requested
+    const hasChangedPage = !equal(itemsPage, prevProps.itemsPage)
+    if (id && itemsPage && hasChangedPage) {
+      const prevPages = prevProps.itemsPage
+      this.fetchCollectionItems(
+        itemsPage && prevProps.itemsPage && Array.isArray(itemsPage) && Array.isArray(prevPages)
```

Comment:
you can simplify this to
```suggestion
        Array.isArray(itemsPage) && Array.isArray(prevPages)
```

right?

Reply:
yes!